File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/channel/ChannelCreateCommand.java
package dev.slne.surf.social.chat.command.channel;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.StringArgument;
import dev.jorel.commandapi.arguments.TextArgument;
import dev.slne.surf.social.chat.object.Channel;
import it.unimi.dsi.fastutil.objects.ObjectArraySet;

public class ChannelCreateCommand extends CommandAPICommand {

  public ChannelCreateCommand(String commandName) {
    super(commandName);

    withRequirement((sender) -> Channel.getChannel(sender) == null);
    withArguments(new TextArgument("name"));
    withArguments(new StringArgument("description"));
    executesPlayer((player, args) -> {
      String name = args.getUnchecked("name");
      String description = args.getUnchecked("description");
      Channel channel = Channel.builder()
          .name(name)
          .description(description)
          .members(new ObjectArraySet<>())
          .invites(new ObjectArraySet<>())
          .closed(true)
          .owner(player)
          .build();

      channel.register();
    });
  }
}

File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/channel/ChannelCreateCommand.kt
package dev.slne.surf.social.chat.command.channel

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.TextArgument
import dev.jorel.commandapi.kotlindsl.getValue
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.social.chat.SurfChat
import dev.slne.surf.social.chat.`object`.Channel
import dev.slne.surf.social.chat.provider.ChannelProvider
import dev.slne.surf.social.chat.util.MessageBuilder

class ChannelCreateCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        withArguments(TextArgument("name"))
        withOptionalArguments(TextArgument("description"))
        playerExecutor { player, args ->
            if (Channel.getChannel(player) != null) {
                SurfChat.send(player, MessageBuilder().error("Du bist bereits in einem Nachrichtenkanal."))
                return@playerExecutor
            }

            val name: String by args
            val description = args.getOrDefaultUnchecked("description", "???")
            val channel = Channel(
                owner = player.uniqueId,
                name = name,
                description = description,
                closed = false
            )

            if (ChannelProvider.exists(name)) {
                SurfChat.send(player, MessageBuilder().error("Der Nachrichtenkanal ").info(name).error(" existiert bereits."))
                return@playerExecutor
            }

            channel.register()
            SurfChat.send(player, MessageBuilder().primary("Du hast den Nachrichtenkanal ").info(channel.name).success(" erstellt."))
        }
    }
}
